src/lib/auth.ts

```diff
@@ -502,17 +519,31 @@ export const createFakeAuthentication = (options: CreateFakeAuthenticationOption
     students: [],
     teachers: [],
   };
-  for (let i = 1; i <= NUM_FAKE_STUDENTS; i++) {
+  //This allows for student groups to be remembered when switching tabs or in cypress tests
+  if (parseInt(userId,10) <= NUM_FAKE_STUDENTS) {
```

Comment:
This test won't work if the random id happens to be "1abc...". I think you'll need a string length test in addition to or instead of the parsed numeric test.

Reply:
Are there other things that will be broken if we don't populate the class with fake students when the id is random?

How about always populating the class with fake students, and then only creating an additional student if it wasn't created before. The simple (inefficient) way to do this is to store the generated ids in an array of strings like `` generatedUserIds.push(`${i}`) `` and then after the loop do a check `if (!generatedUserIds.includes(userId)) { ...`